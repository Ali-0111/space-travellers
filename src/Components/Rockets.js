const Rockets = () => (
  <div>
    This is Rockets page.
  </div>
);

export default Rockets;
