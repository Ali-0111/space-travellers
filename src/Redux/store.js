import { configureStore, combineReducers } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketSlice from './rockets/rocketsSlice';

const rootReducer = combineReducers({
  missionsReducer: missionsSlice.reducer,
  reservationListKey: rocketSlice.reducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
