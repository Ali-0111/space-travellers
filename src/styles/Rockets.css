.reservation-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 15px;
  padding: 5px;
}
