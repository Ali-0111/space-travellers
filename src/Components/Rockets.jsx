import React from 'react';
import { useSelector } from 'react-redux';
import '../styles/Rockets.css';

const Rockets = () => {
  const storedData = useSelector((store) => (store.reducer.reservationList));
  return (
    <section className="reservation-wrapper">
      <article>
        <h2>Rocket data will be dispalyed </h2>
      </article>
    </section>
  );
};

export default Rockets;
