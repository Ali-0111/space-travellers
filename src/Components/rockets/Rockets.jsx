import React from 'react';
import { useSelector } from 'react-redux';
import ArticleComponent from './RocketsArticle';
import '../../styles/Rockets.css';

const Rockets = () => {
  const stateData = useSelector((store) => (store.reservationListKey.reservationList));

  return (
    <section className="reservation-wrapper">
      {
        stateData.map((article) => (
          <ArticleComponent
            key={article.id}
            articleData={article}
          />
        ))
      }
    </section>
  );
};

export default Rockets;
