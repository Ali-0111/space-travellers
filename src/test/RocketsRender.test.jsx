import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import store from '../Redux/store';

import Rockets from '../Components/rockets/Rockets';

describe('Rockets rendering Validation', () => {
  test('Rockets should render with / url', () => {
    const treeSnapShot = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Rockets />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(treeSnapShot).toMatchSnapshot();
  });
});
