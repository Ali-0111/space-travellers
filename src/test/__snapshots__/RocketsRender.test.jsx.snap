// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets rendering Validation Rockets should render with / url 1`] = `
<section
  className="reservation-wrapper"
/>
`;
